transformers/avayaVoiceChannel/transformer.ts

```diff
@@ -0,0 +1,231 @@
+
+	/**
+	 * This transformer is executed when receiving a message
+	 * from the user, before executing the Flow.
+	 *
+	 * @param endpoint The configuration object for the used Endpoint.
+	 * @param request The Express request object with a JSON parsed body.
+	 * @param response The Express response object.
+	 *
+	 * @returns A valid userId, sessionId, as well as text and/or data,
+	 * which has been extracted from the request body.
+	 */
+	handleInput: async ({ endpoint, request, response }) => {
+
+		/**
+		 * Extract the userId, sessionId and text
+		 * from the request. Example:
+		 *
+		 * const { userId, sessionId, text, data } = request.body;
+		 *
+		 * Note that the format of the request body will be different for
+		 * every Endpoint, and the example above needs to be adjusted
+		 * accordingly.
+		 */
+		const { body } = request as CPaaSRequest;
+		console.log('body='.concat(JSON.stringify(body)));
+		const { From, To, CallSid, Digits, SpeechResult, SmsSid, Body } = body;
+		const userId = From;
+		const sessionId = CallSid ? CallSid : SmsSid;
+		let sessionStorage = await getSessionStorage(userId,sessionId);
+		sessionStorage.From = From;
+		sessionStorage.To = To;
+		sessionStorage.cpaas_channel = CallSid ? 'call' : 'sms';
+		console.log('handleInput: sessionStorage.cpaas_channel='.concat(sessionStorage.cpaas_channel));
+		let data = {"call": false,"sms":false, "phone":From};
+		const menu = sessionStorage['menu'] ? sessionStorage['menu'] : {};
+		let text = '';
+		switch (sessionStorage.cpaas_channel) {
+			case 'call':
+				text = Digits ? (menu[Digits] ? menu[Digits] : 'default') : SpeechResult;
+				data.call = true;
+				break;
+			case 'sms':
+				text = Body;
+				data.sms = true;
+				break;
+			default:
+				text = 'default';
+				break;
+		}
+		console.log('input.text='.concat(text));
+		return {
+			userId,
+			sessionId,
+			text,
+			data
+		};
+	},
+
+	/**
+	 * This transformer is executed on every output from the Flow.
+	 * 
+	 * @param output The raw output from the Flow. It is possible to manipulate
+	 * and return every distinct output before they get formatted in the 'handleExecutionFinished'
+	 * transformer.
+	 *
+	 * @param endpoint The configuration object for the used Endpoint.
+	 * @param userId The unique ID of the user.
+	 * @param sessionId The unique ID for this session. Can be used together with the userId
+	 * to retrieve the sessionStorage object.
+	 * 
+	 * @returns The output that will be formatted into the final response in the 'handleExecutionFinished' transformer.
+	 */
+	handleOutput: async ({ output, endpoint, userId, sessionId }) => {
+		console.log('handleOutput called');
+        const sessionStorage = await getSessionStorage(userId, sessionId);
+		if (output.data != null && output.data._cognigy != null) {
+			const payload = output.data._cognigy;
+			console.log('payload='.concat(payload));
+			const activities = payload._spoken.json.activities;
+			activities.forEach( (a) => {
```

Comment:
Please call it *activity* and not *a* in the forEach loop.

Reply:
made the changes per suggestion